src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,90 @@
+import React, {useState, useEffect} from 'react';
+import './Pagination.css'
+
+interface PaginationProps {
+  data: any[];
+  RenderComponent: React.FC<any>;
+  title?: string;
+  pageLimit: number;
+  dataLimit: number;
+}
+
+
+function Pagination ({data, RenderComponent, title, pageLimit, dataLimit}: PaginationProps) {
+  const [pages] = useState(Math.round(data.length / dataLimit));
+  const [currentPage, setCurrentPage] = useState(1);
+
+  useEffect(() => {
+    window.scrollTo({ behavior: 'smooth', top: 0 });
+  }, [currentPage]);
+
+  function goToNextPage() {
+    setCurrentPage((page) => page + 1);
+  }
+
+  function goToPreviousPage() {
+    setCurrentPage((page) => page - 1);
+  }
+
```

Comment:
Both gotonextpage and gotopreviouspage can be compacted into a single faction with a parameter, another thing that can be refactored at some point

Reply:
this is fine for now, but this feature should be an 'infinite scrolling' feature instead of a page system, 

for this you will need to look into 'lazy loading' basics of web development & react

1) we will keep the results, aka 'cache' it 
2) 'lazy load' the 'page' as the person scrolls

@cguizar25